### PR TITLE
NMS-14249: Allow other Oids for BgpSessionMonitor

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://issues.opennms.org/browse/NMS-14249

### CODE REVIEW

File: features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/BgpSessionMonitor.java [hunk 1]
```diff
@@ -55,34 +55,67 @@
...
      */
-    private static final String BGP_PEER_FSM_EST_TIME_OID = ".1.3.6.1.2.1.15.3.1.16";
-    
+    private static final String BGP_PEER_LAST_ERROR_SUFFIX = ".3.1.14";
+    /**
+     * Suffix for the table that represents the BGP-peer established time.
+     */
+    private static final String BGP_PEER_FSM_EST_TIME_SUFFIX = ".3.1.16";
+
+    class BgpOids {
```
Comment (fooker) [inline features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/BgpSessionMonitor.java:86]: can be `private static`?

File: features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/BgpSessionMonitor.java [hunk 2]
```diff
@@ -117,6 +150,10 @@ private int value() {
         }
     };
 
+    BgpOids constructOids(final Map<String, Object> parameters) {
+        return new BgpOids(parameters);
+    }
```
Comment (fooker) [inline features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/BgpSessionMonitor.java:155]: inline this constructor call

**Review by fooker — CHANGES_REQUESTED** [review]

File: features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/BgpSessionMonitor.java (hunk 1 above)
Comment (christianpape) [reply features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/BgpSessionMonitor.java:86]: done, but static only due to tests.

File: features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/BgpSessionMonitor.java (hunk 2 above)
Comment (christianpape) [reply features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/BgpSessionMonitor.java:155]: done.

**Review by fooker — APPROVED** [review]

File: docs/modules/reference/pages/service-assurance/monitors/BgpSessionMonitor.adoc [hunk 3]
```diff
@@ -74,6 +74,30 @@ See http://www.freesoft.org/CIE/RFC/1771/31.htm[RFC1771 BGP Finite State Machine
 | bgpPeerIp
 | IP address of the far end BGP peer session
 | n/a
+
+| bgpBaseOid
+| Default base OID for the table that represents the BGP sessions
+| .1.3.6.1.2.1.15
+
+| bgpPeerStateOid
+| OID suffix for the table that represents the BGP-peer states
```
Comment (Bonrob2) [inline docs/modules/reference/pages/service-assurance/monitors/BgpSessionMonitor.adoc:83]:
```suggestion
| OID suffix for the table that represents the BGP-peer states.

```

File: docs/modules/reference/pages/service-assurance/monitors/BgpSessionMonitor.adoc [hunk 4]
```diff
@@ -74,6 +74,30 @@ See http://www.freesoft.org/CIE/RFC/1771/31.htm[RFC1771 BGP Finite State Machine
...
+| bgpPeerStateOid
+| OID suffix for the table that represents the BGP-peer states
+| bgpBaseOid.3.1.2
+
+| bgpPeerAdminStateOid
+| OID suffix for the table that represents the BGP-peer admin states
+| bgpBaseOid.3.1.3
+
+| bgpPeerRemoteAsOid
+| OID suffix for the table that represents the BGP-peer remote AS number
```
Comment (Bonrob2) [inline docs/modules/reference/pages/service-assurance/monitors/BgpSessionMonitor.adoc:91]:
```suggestion
| OID suffix for the table that represents the BGP-peer remote AS number.

```

File: docs/modules/reference/pages/service-assurance/monitors/BgpSessionMonitor.adoc [hunk 5]
```diff
@@ -74,6 +74,30 @@ See http://www.freesoft.org/CIE/RFC/1771/31.htm[RFC1771 BGP Finite State Machine
...
+| bgpBaseOid.3.1.9
+
+| bgpPeerLastErrorOid
+| OID suffix for the table that represents the BGP-peer last error code
```
Comment (Bonrob2) [inline docs/modules/reference/pages/service-assurance/monitors/BgpSessionMonitor.adoc:95]:
```suggestion
| OID suffix for the table that represents the BGP-peer last error code.

```

File: docs/modules/reference/pages/service-assurance/monitors/BgpSessionMonitor.adoc [hunk 6]
```diff
@@ -74,6 +74,30 @@ See http://www.freesoft.org/CIE/RFC/1771/31.htm[RFC1771 BGP Finite State Machine
...
+| bgpBaseOid.3.1.14
+
+| bgpPeerFsmEstTimeOid
+| OID suffix for the table that represents the BGP-peer established time
```
Comment (Bonrob2) [inline docs/modules/reference/pages/service-assurance/monitors/BgpSessionMonitor.adoc:99]:
```suggestion
| OID suffix for the table that represents the BGP-peer established time.

```

**Review by fooker — APPROVED** [review]

**Review by Bonrob2 — APPROVED** [review]
docs look good.

**Pull request merged**